docs(linter): Add a few more stylistic react rules to NOT_SUPPORTED_RULE_NAMES.

connorshea opened the pull request:
These rules are stylistic and would conflict with oxfmt/prettier-based formatting. As such, we should not implement them.

These additional rules were sourced from the eslint-config-prettier lib https://github.com/prettier/eslint-config-prettier/blob/07829b4912d173986610a4985247896b09f9fcaf/index.js#L197

This should hopefully finish up the list of not-supported react rules for good, and we won't need to add any more.

camc314 commented:
### Merge activity

* **Dec 30, 6:29 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 30, 6:29 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
docs(linter): Add a few more stylistic react rules to NOT_SUPPORTED_RULE_NAMES. (#17505)

These rules are stylistic and would conflict with oxfmt/prettier-based formatting. As such, we should not implement them.

These additional rules were sourced from the eslint-config-prettier lib https://github.com/prettier/eslint-config-prettier/blob/07829b4912d173986610a4985247896b09f9fcaf/index.js#L197

This should hopefully finish up the list of not-supported react rules for good, and we won't need to add any more.
